src/test/SubscribableRequestQueue.test.ts

```diff
@@ -0,0 +1,262 @@
+import { expect } from "chai";
+
+import SubscribableRequestQueue from "../utils/SubscribableRequestQueue";
+
+const TIMEOUT = 10;
+const LONG_TIMEOUT = 20;
+
+/** Returns a promise that resolves after `timeoutMs` milliseconds with value `result`. */
+function delay<T>(timeoutMs: number, result: T): Promise<T> {
+  return new Promise((resolve) => setTimeout(() => resolve(result), timeoutMs));
+}
+
+/** Returns a promise that rejects after `timeoutMs` milliseconds with reason `reason`. */
+function delayReject(timeoutMs: number, reason: unknown): Promise<unknown> {
+  return new Promise((_, reject) => setTimeout(() => reject(reason), timeoutMs));
+}
+
+/** Determines whether a promise has rejected. */
+// From https://gist.github.com/tyru/29360dfa475d2fefaf6c4655a93c2cb0
+function isRejected(promise: Promise<unknown>): Promise<boolean> {
+  return Promise.race([
+    delay(0, false),
+    promise.then(
+      () => false,
+      () => true
+    ),
+  ]);
+}
+
+describe("SubscribableRequestQueue", () => {
+  describe("addSubscriber", () => {
+    it("adds a subscriber", () => {
+      const queue = new SubscribableRequestQueue();
+      const id = queue.addSubscriber();
+      expect(queue.hasSubscriber(id)).to.be.true;
+    });
+
+    it("returns unique ids", () => {
+      const queue = new SubscribableRequestQueue();
+      const id1 = queue.addSubscriber();
+      const id2 = queue.addSubscriber();
+      expect(id1).to.not.equal(id2);
```

Comment:
I'd add one more test to remove one of id1 or id2, and then add an id3, and confirm it's also unique.